### PR TITLE
ENH/BUG: inst_id and tag check

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #784, #537

Adds a check if the user supplied `tag` and `inst_id` are actually supported by the requested instrument support module.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for
your test configuration

- Added a test, with error message check, when an invalid tag or inst_id provided

**Test Configuration**:
* Operating system: MacOS 10.15.4
* Version number: Python 3.7

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/_instrument.py [hunk 1]
```diff
@@ -1193,6 +1200,22 @@ def _assign_attrs(self, by_name=False, inst_module=None):
             # No module or name info, default pass functions assigned
             return
 
+        # Check if tag and inst_id are appropriate for the module
+        if inst_id not in inst.inst_ids.keys():
+            inst_id_str = ' '.join([''.join(('"', key, '"'))
+                                    for key in inst.inst_ids.keys()])
```
Comment (aburrell) [inline pysat/_instrument.py:1206]:
```suggestion
            inst_id_str = ' '.join([''.join(('"', ikey, '"'))
                                    for ikey in inst.inst_ids.keys()])
```

File: pysat/_instrument.py [hunk 2]
```diff
@@ -1193,6 +1200,22 @@ def _assign_attrs(self, by_name=False, inst_module=None):
...
+            estr = ''.join(('"', inst_id, '" is not one of the supported ',
+                            'inst_ids. Supported inst_ids are: ',
+                            inst_id_str, '.'))
+            raise ValueError(estr)
+
+        if tag not in inst.inst_ids[inst_id]:
+            tag_str = ' '.join([''.join(('"', key, '"'))
+                                for key in inst.inst_ids[inst_id]])
```
Comment (aburrell) [inline pysat/_instrument.py:1214]:
```suggestion
            tag_str = ' '.join([''.join(('"', tkey, '"'))
                                for tkey in inst.inst_ids[inst_id]])
```

File: pysat/tests/test_files.py [hunk 3]
```diff
@@ -780,7 +780,6 @@ def test_files_non_standard_pysat_directory(self):
         self.testInst = \
             pysat.Instrument(inst_module=pysat.instruments.pysat_testing,
```
Comment (aburrell) [inline pysat/tests/test_files.py:781]: Recommend breaking this line on the open paren.

File: pysat/tests/test_instrument.py [hunk 4]
```diff
@@ -2558,6 +2558,27 @@ def test_incorrect_creation_empty_instrument_object(self):
         estr = 'Inputs platform and name must both'
         assert str(err).find(estr) >= 0
 
+    def test_error_bad_inst_id_instrument_object(self):
+        """Ensure instantiation with invalid inst_id errors"""
+        with pytest.raises(ValueError) as err:
+            # both name and platform should be empty
+            _ = pysat.Instrument(platform=self.testInst.platform,
+                                 name=self.testInst.name,
+                                 inst_id='invalid_inst_id')
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:2567]:
```suggestion
            pysat.Instrument(platform=self.testInst.platform,
                             name=self.testInst.name,
                             inst_id='invalid_inst_id')
```

File: pysat/tests/test_instrument.py [hunk 5]
```diff
@@ -2558,6 +2558,27 @@ def test_incorrect_creation_empty_instrument_object(self):
...
+        assert str(err).find(estr) >= 0
+
+    def test_error_bad_tag_instrument_object(self):
+        """Ensure instantiation with invalid inst_id errors"""
+        with pytest.raises(ValueError) as err:
+            # both name and platform should be empty
+            _ = pysat.Instrument(platform=self.testInst.platform,
+                                 name=self.testInst.name,
+                                 inst_id='',
+                                 tag='bad_tag')
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:2578]:
```suggestion
            pysat.Instrument(platform=self.testInst.platform,
                             name=self.testInst.name,
                             inst_id='', tag='bad_tag')
```

**Review by aburrell — COMMENTED** [review]
Style suggestions.

**Comment by aburrell** [comment]
Following the test in #784 I get: `ValueError: "fun" is not one of the supported inst_ids. Supported inst_ids are: "".`.  This is a timely and informative error 🦉 

Altering the test to use `inst_id=''`, I get: `ValueError: "hi" is not one of the supported tags. Supported tags are: "realtime" "historic".`

I think the join strings should include commas to make these messages easier to read.

File: pysat/_instrument.py (hunk 1 above)
Comment (aburrell) [reply pysat/_instrument.py:1206]: Another alternative is to not define `inst_id_str` and `tag_str` here, but instead use: `list(inst.inst_ids.tags.keys()).__repr__()`

Comment (rstoneback) [reply pysat/_instrument.py:1206]: I merged your suggestions together.

**Review by aburrell — APPROVED** [review]

**Pull request merged**